vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc

```diff
@@ -43,7 +43,9 @@ LedgerImpl::LedgerImpl(LedgerClient* client)
   set_ledger_client_for_logging(ledger_client_);
 }
 
-LedgerImpl::~LedgerImpl() = default;
+LedgerImpl::~LedgerImpl() {
+  set_ledger_client_for_logging(nullptr);
```

Comment:
It was a possible use-after-free in case of using logging after this point (or in the same time on another thread)